presto-main/src/main/java/io/prestosql/execution/DataDefinitionTask.java

```diff
@@ -33,11 +32,21 @@
 
     default String explain(T statement, List<Expression> parameters)
     {
-        if (statement instanceof Prepare) {
-            return SqlFormatter.formatSql(statement, Optional.empty());
+        StringBuilder builder = new StringBuilder();
+
+        builder.append(SqlFormatter.formatSql(statement));
+
+        if (!parameters.isEmpty()) {
+            String formatted = parameters.stream()
+                    .map(Expression::toString)
+                    .collect(Collectors.joining(", ", "[", "]"));
```

Comment:
formatted eq parameters.toString()

(List#toString is documented, stable, AFAIK)

in fact, you don't use `formatted` at all

(also, static import .joining)

Reply:
Left-over from some work-in-progress...